docs/extensions/language-support.md

```diff
@@ -0,0 +1,861 @@
+
+# VS Code Language Extensions
+When you hear about a language being supported in VS Code, you
+usually think first of syntax highlighting, code completion, and if applicable
+debugging support. This is a good start, but language extensions can do a lot more.
+
+By providing configuration files, an extension can support syntax highlighting,
+snippets, and some smart editing features. For more advanced features your extension needs to
+do real work. We recommend that you implement a language server your extension talks to.
+
+A language server is a stand-alone server that speaks the
+[language server protocol](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md).
+You can implement the server in the language that is best suited for the task. For example,
+if there are good libraries written in Python for the language you want to support, you
+might want to consider implementing your language server in Python. If you choose to implement your
+language server in JavaScript or TypeScript you can build on top of our
+[npm modules](https://github.com/Microsoft/vscode-languageserver-node).
+
+Besides the implementation language, you have flexibility in deciding which parts of the
+[language server protocol](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md)
+your language server implements. The only thing you have to make sure is that your server
+correctly announces its capabilities in response to the `initialize` method.
+
+However, the language server architecture is not the only possible way to implement your
+extension. You can also implement the various language support features directly in your extension.
+
+In order to make it easier for you to decide what to implement first and what to improve later on
+we list the various support features below with examples of how they are exposed to users and to which
+classes and methods or language server protocol messages they map to.
+
+After that we'll list some methods that are available to manager the state in your extension.
+
+
```

Comment:
I'd add a short paragraph commenting that this is the configuration section that only requires TextMate files.